Kconfig.zephyr

```diff
@@ -812,15 +812,6 @@ config BUILD_OUTPUT_INFO_HEADER
 	  - VMA address of each segment
 	  - Size of each segment
 
-config BUILD_ALIGN_LMA
-	bool "Align LMA in output image"
-	default y if BUILD_OUTPUT_ADJUST_LMA!=""
-	help
-	  Ensure that the LMA for each section in the output image respects
-	  the alignment requirements of that section. This is required for
-	  some tooling, such as objcopy, to be able to adjust the LMA of the
-	  ELF file.
```

Comment:
Given that this was part of 3.5 and 3.6 releases, a proper deprecation would be necessary.

Also this might still be useful for the users relying on third-party GNU toolchains (e.g. GNU Arm Embedded) because the [binutils patch](https://github.com/zephyrproject-rtos/binutils-gdb/pull/12) is Zephyr SDK-specific.

Reply:
That's a good point. I'm totally okay to leave this in tree, this was just in response to @fgrandel's request here: https://github.com/zephyrproject-rtos/zephyr/issues/58080#issuecomment-1768645648. There were also other issues that this workaround introduced, see here: https://github.com/zephyrproject-rtos/zephyr/pull/62369

Do you think we should deprecate the Kconfig out?

Reply:
@stephanosio what do you want @danieldegrasse to do here?